src/Symfony/Component/BrowserKit/Test/Constraint/BrowserCookieValueSame.php

```diff
@@ -34,9 +34,8 @@ public function toString(): string
         if ($this->domain) {
             $str .= \sprintf(' for domain "%s"', $this->domain);
         }
-        $str .= \sprintf(' with %svalue "%s"', $this->raw ? 'raw ' : '', $this->value);
 
-        return $str;
```

Comment:
should be reverted

Reply:
Oh alright, I missed the exact point of "All those .= or += are perf optimizations.". Should be good now, I also reverted all casters

Reply:
It's very debatable if we need such a perf optimization in a test class in my opinion. There are thousands of concatenating operators in the Symfony code base that don't do it in place.

To be honest, all of the ones you commented on do not look like to me they are on the critical path of performance: it's either dev tools (browserkit, vardumper) or actions that are not in a critical path (the impersonator url generator might be the only one that we can argue might be called on every request).

Reply:
CS shouldn't be a reason to degrade perf (var-dumper is perf-critical, everybody wants fast dumps). That's also not worth the future merge conflicts. I don't see any upsides that outweighs the downsides.